cloud/cluster/memory/cluster.go

```diff
@@ -0,0 +1,138 @@
+package memory
```

Comment:
I agree you could do it this way, but then everyone who implements Cluster logic has to reimplement all this core subscription logic which seems wasteful.  If we truly want this to be consumed as an OSS project, we should provide as much core logic as possible and then just have people plug in their infrastructure specific things like service discovery.  That's why it was originally designed that way where DynamicCluster contained core logic on how to do this and the you could just notify it when nodes were added or removed.